### PR TITLE
Add support for observations contained in place.

### PR DESCRIPTION
**Pull request by keyurva**
* I think the expression support can be extended / generalized to support more types of expressions but sticking with V2 API equivalency for now so only supporting contained in place.

### CODE REVIEW

File: internal/server/spanner/datasource.go [hunk 1]
```diff
@@ -68,15 +68,31 @@ func (sds *SpannerDataSource) Node(ctx context.Context, req *v3.NodeRequest) (*v
...
-	if err != nil {
-		return nil, fmt.Errorf("error getting observations: %v", err)
+	var observations []*Observation
+
+	if entityExpr != "" {
+		containedInPlace, err := v2.ParseContainedInPlace(entityExpr)
+		if err != nil {
+			return nil, fmt.Errorf("error getting observations (contained in): %v", err)
+		}
+		observations, err = sds.client.GetObservationsContainedInPlace(ctx, variables, containedInPlace)
```
Comment (n-h-diaz) [inline internal/server/spanner/datasource.go:81]: Were there any performance concerns with fetching the full series for all the children places?

**Review by n-h-diaz — APPROVED** [review]
Thanks for adding!

File: internal/server/spanner/datasource.go (hunk 1 above)
Comment (keyurva) [reply internal/server/spanner/datasource.go:81]: For the golden tests it was fine. For something like "Count_Person" and "country/USA<-containedInPlace+{typeOf:State}" it was noticeable.

**Comment by keyurva** [comment]
Thanks for the review!

**Pull request merged**